packages/api/src/controllers/admin.js

```diff
@@ -0,0 +1,172 @@
+import { authMiddleware } from '../middleware'
+import Router from 'express/lib/router'
+import { db } from '../store'
+import sql from 'sql-template-strings'
+import consul from 'consul'
+
+// const consul = require('consul')()
+
+const ACTIVE_TIMEOUT = 1 * 3600 * 1000 // 1h
+
+const app = Router()
+const traefikKeyPathRouters = `traefik/http/routers/`
+const traefikKeyPathServices = `traefik/http/services/`
+const traefikKeyPathMiddlewares = `traefik/http/middlewares/`
+
+app.post(
+  '/consul-routes-clean',
+  authMiddleware({ anyAdmin: true }),
+  async (req, res) => {
+    // check if there is routes in the consul that belongs to the streams that
+    // are not active anymore
+    let baseUrl
+    if (req.config.consul) {
+      baseUrl = req.config.consul
+      if (!baseUrl.endsWith('/v1')) {
+        baseUrl += '/v1'
+      }
+    }
+    const cc = consul({ promisify: true, baseUrl })
+    let keys
+    try {
+      keys = await cc.kv.keys({
+        key: traefikKeyPathRouters,
+        separator: '/',
+      })
+      keys = keys.map((k) => k.split('/')[3])
+    } catch (e) {
+      if (e.statusCode == 404) {
+        return res.json({ msg: 'no keys' })
+      } else {
+        console.log(`error getting keys:`, e)
+        res.statusCode = 500
+        return res.json({ error: String(e) })
```

Comment:
I like that we're mostly consistent on returning errors as an array:
```suggestion
        return res.json({ errors: [String(e)] })
```

Reply:
fixed